### PR TITLE
[Server] JPEG output for WMS GetPrint request has gone

### PR DESCRIPTION
**Pull request by rldhont**
## Description

In QGIS Server 2.* the WMS GetPrint request could genrate JPEG image. QGIS Server 3.4 has lost this capabilities.
```
<ServiceExceptionReport xmlns="http://www.opengis.net/ogc" version="1.3.0" capture-installed="true">
    <ServiceException code="InvalidFormat">
        Output format jpg is not supported by the GetPrint request
    </ServiceException>
</ServiceExceptionReport>
```
To fix this regression, it is necessary to accept JPEG output format.

And to avoid this regression to come back, the QGIS Server tests has been updated to accept jpg image test.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/services/wms/qgswmsgetprint.cpp [hunk 1]
```diff
@@ -39,6 +39,10 @@ namespace QgsWms
     {
       contentType = "image/png";
     }
+    else if ( format == QgsWmsParameters::JPG )
+    {
+      contentType = "image/jpeg";
```
Comment (elpaso) [inline src/server/services/wms/qgswmsgetprint.cpp]: If `contentType` is a `QString` you should use `QStringLiteral("image/jpeg");` also for the others.

**Review by elpaso — APPROVED** [review]
Cool, thanks!

File: src/server/services/wms/qgswmsgetprint.cpp [hunk 2]
```diff
@@ -39,6 +39,10 @@ namespace QgsWms
     {
       contentType = "image/png";
     }
+    else if ( format == QgsWmsParameters::JPG )
```
Comment (nyalldawson) [inline src/server/services/wms/qgswmsgetprint.cpp]: This should really be a switch to get warnings and avoid bugs like this

File: src/server/services/wms/qgswmsgetprint.cpp (hunk 1 above)
Comment (rldhont) [reply src/server/services/wms/qgswmsgetprint.cpp]: Changes done

File: src/server/services/wms/qgswmsgetprint.cpp (hunk 2 above)
Comment (rldhont) [reply src/server/services/wms/qgswmsgetprint.cpp]: Changes done

**Comment by rldhont** [comment]
@elpaso @pblottiere I have fixed the tests, thanks to kindly review the code.

**Comment by pblottiere** [comment]
@rldhont Thanks, it looks good now :+1:!

**Review by pblottiere — APPROVED** [review]

**Pull request merged**